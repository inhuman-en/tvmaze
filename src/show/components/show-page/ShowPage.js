import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import './ShowPage.scss';

import { loadShowDetails } from '../../store';

import ShowDetails from '../show-details/ShowDetails';

const ShowPage = () => {
  const { id } = useParams();
  const {details, error } = useSelector(state => state.show);
  const dispatch = useDispatch();

  useEffect(
    () => dispatch(loadShowDetails(id)),
    [id, dispatch],
  );

  if (error) {
    return error;
  }

  if (!details) {
    return <div>Loading...</div>
  }

  return <ShowDetails data={details} />;
};

export default ShowPage;
