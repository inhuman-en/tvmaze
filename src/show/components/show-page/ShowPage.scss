@import 'style/variables';

$nav__height: 30px;
$padding: 15px;

.show-page {
  &__nav {
    display: flex;
    justify-content: center;
    align-items: center;
    height: $nav__height;
    & a {
      padding: 5px;
      font-weight: $semibold;
      &:hover {
        text-decoration: underline;
      }
    }
  }
}

