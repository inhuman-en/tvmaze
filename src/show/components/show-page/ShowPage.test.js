import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ShowPage from './ShowPage';

configure({ adapter: new Adapter() });

jest.mock('../../hooks');

jest.mock('react-router-dom', () => ({
  useParams: () => ({ id: 2 }),
  useRouteMatch: () => ({ path: '/show/2' }),
  generatePath: jest.fn(),
  Link: ({ children }) => <div>{children}</div>,
}));

describe('ShowPage component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render', () => {
    require('react-router-dom').generatePath.mockReturnValueOnce(1).mockReturnValueOnce(3);

    const result = toJson(shallow(<ShowPage />));
    expect(result).toMatchSnapshot();
  });
});
