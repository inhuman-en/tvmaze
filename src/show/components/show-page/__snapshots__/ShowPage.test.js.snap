// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowPage component should render 1`] = `
<Fragment>
  <nav
    className="show-page__nav"
  >
    <Link
      to={1}
    >
      &lt;Prev
    </Link>
    <Link
      to={3}
    >
      Next&gt;
    </Link>
  </nav>
  <div
    className="show-page__main"
  >
    <ShowDetails
      data={
        Object {
          "id": 2,
          "name": "show name",
          "summary": "show summary",
        }
      }
    />
    <EpisodeList
      data={Array []}
    />
  </div>
</Fragment>
`;
